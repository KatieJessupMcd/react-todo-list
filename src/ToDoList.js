import React, { Component } from 'react';
import NewToDoForm from './NewToDoForm';
import Todo from './Todo';
import uuid from 'uuid/v4';
import EditToDoForm from './EditToDoForm';

class ToDoList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      tasks: [],
      currentlyEdited: '',
      currentlyEditedId: '',
      edit: false
    };
    this.addTodo = this.addTodo.bind(this);
    this.updateTodo = this.updateTodo.bind(this);
    this.removeTodo = this.removeTodo.bind(this);
    this.showEditForm = this.showEditForm.bind(this);
  }

  renderTasks() {
    return (
      <ol id="to-do-list">
        {this.state.tasks.map(t => (
          <li key={t.id}>
            <Todo
              id={t.id}
              task={t.task}
              remove={this.removeTodo}
              edit={this.showEditForm}
            />
            {/*  Conditional, if edit button is clicked, render EditTodo */}
          </li>
        ))}
      </ol>
    );
  }

  addTodo(task) {
    let newTask = { ...task, id: uuid() };
    this.setState(currState => ({
      tasks: [...currState.tasks, newTask]
    }));
  }

  updateTodo(task, taskId) {
    // Use the taskId passed through to update the task
    // get the task name from edit form's state
    this.setState(currState => ({
      tasks: currState.tasks.map(function(t) {
        if (t.id === taskId) {
          t.task = task;
        }
        return t;
      })
    }));
    this.setState({ edit: false });
  }

  removeTodo(taskId) {
    this.setState(currState => ({
      tasks: currState.tasks.filter(t => t.id !== taskId)
    }));
  }

  showEditForm(taskId, taskName) {
    // this should display an edit form
    this.setState({
      edit: true,
      currentlyEdited: taskName,
      currentlyEditedId: taskId
    });
  }

  render() {
    if (this.state.edit) {
      return (
        <div className="TodoList">
          To-Do List
          <NewToDoForm addTodo={this.addTodo} />
          <EditToDoForm
            updateTodo={this.updateTodo}
            currentTaskName={this.state.currentlyEdited}
            currentTaskId={this.state.currentlyEditedId}
          />
          {this.renderTasks()}
        </div>
      );
    }

    return (
      <div className="TodoList">
        To-Do List
        <NewToDoForm addTodo={this.addTodo} />
        {this.renderTasks()}
      </div>
    );
  }
}

export default ToDoList;
