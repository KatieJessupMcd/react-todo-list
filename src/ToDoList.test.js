import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ToDoList from './ToDoList';

// smoke test
it('renders without crashing', function() {
  shallow(<ToDoList />);
});

// snapshot test
it('matches snapshot', function() {
  let wrapper = shallow(<ToDoList />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
