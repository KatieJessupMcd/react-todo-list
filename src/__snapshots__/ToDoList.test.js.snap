// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="TodoList"
>
  To-Do List
  <NewToDoForm
    addTodo={[Function]}
  />
  <ol
    id="to-do-list"
  />
</div>
`;
